pybloqs/block/base.py

```diff
@@ -83,25 +47,52 @@ def __init__(self, title=None, title_level=3, title_wrap=False,
     def render_html(self, pretty=True, static_output=False, header_block=None, footer_block=None):
         """Returns html output of the block
         :param pretty: Toggles pretty printing of the resulting HTML. Not applicable for non-HTML output.
+        :param static_output: Passed down to _write_block. Will render static version of blocks which support this.
+        :param header_block: If not None, header is inlined into a HTML body as table.
+        :param footer_block: If not None, header is inlined into a HTML body as table.
         :return html-code of the block
         """
         # Render the contents
         html = root("html", doctype="html")
         head = append_to(html, "head")
-        head = append_to(head, "meta", charset='utf-8')
+        append_to(head, "meta", charset='utf-8')
+
         body = append_to(html, "body")
 
         # Make sure that the main style sheet is always included
         resource_deps = DependencyTracker(default_css_main)
-        if header_block is not None:
-            header_block._write_block(body, Cfg(), id_generator(), resource_deps=resource_deps,
-                                      static_output=static_output)
-
-        self._write_block(body, Cfg(), id_generator(), resource_deps=resource_deps, static_output=static_output)
 
-        if footer_block is not None:
-            footer_block._write_block(body, Cfg(), id_generator(), resource_deps=resource_deps,
-                                      static_output=static_output)
+        # If header or footer are passed into this function, inline them in the following structure:
+        #
+        # <body>
+        # <table>
+        #    <thead><tr><td>Header html</td></tr></thead>
+        #    <tfoot><tr><td>Footer html</td></tr></tfoot>
+        #    <tbody><tr><td>Body html</td></tr></tbody>
+        # </table>
+        # </body>
+        if header_block is not None or footer_block is not None:
```

Comment:
wouldn't the if statements be more succinct with truthy/falsey ? i.e.
```python
if header_block or footer_block:
    ...
    if header_block:
         ...
    if footer_block:
         ...
else:
    ...      
```

Reply:
Hey Han, Thank you very much for reviewing!
I am going with PEP-8 here: "Comparisons to singletons like None should always be done with is or is not, never the equality operators."
See https://www.python.org/dev/peps/pep-0008/#programming-recommendations

Reply:
is the header_block and footer_block not always a `str` ?

Reply:
In an ideal use case, these should be BaseBlock objects (or inherited from there). So it would make the implicit assumption that BaseBlock and all of its children never evaluate False.

Reply:
in that case, it makes sense to use the explicit is None comparisons.